System/System.cpp

```diff
@@ -34,7 +49,87 @@ namespace RTE {
 #endif
 	}
 
+	void System::Destroy() {
+#if defined(__unix__) && !LINUX_PORTABLE
+		std::filesystem::remove_all(s_TempDirectory);
+#endif
+	}
+
 /////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+#ifdef __unix__
+	std::filesystem::path System::GetXdgDataHome() {
+		char *envXdgDataHome = std::getenv("XDG_DATA_HOME");
+
+		if (envXdgDataHome) {
+			return std::filesystem::path(envXdgDataHome) / std::filesystem::path("Cortex Command");
+		} else {
+			char *envUserHome = std::getenv("HOME");
+			if (!envUserHome) {
+				struct passwd *pw = getpwuid(getuid());
+				envUserHome = pw->pw_dir;
+			}
+			return envUserHome / std::filesystem::path(".local/share/Cortex Command/");
+		}
+	}
+
+	void System::SetupBaseGameFolders(const std::filesystem::path &tempDirectory) {
```

Comment:
If you restructure `System::Initialize` so this and the other one are called after `s_WorkingDirectory` is assigned to you can eliminate the params and just use `s_WorkingDirectory` instead.

And even if you don't, you don't need them anyway cause you have `s_TempDirectory` (which you don't need cause you're setting the working dir to it and `s_WorkingDirectory` ends up being the same dir).